packages/app/src/cli/commands/app/generate/schema.ts

```diff
@@ -0,0 +1,42 @@
+import {Flags} from '@oclif/core'
+import Command from '@shopify/cli-kit/node/base-command'
+import {path, error, cli, output} from '@shopify/cli-kit'
+import {appFlags} from '../../../flags.js'
+import {AppInterface} from '../../../models/app/app.js'
+import {load as loadApp} from '../../../models/app/loader.js'
+import {generateSchemaService} from '../../../services/generate-schema.js'
+
+export default class GenerateSchema extends Command {
+  static description = 'Generates a GraphQL schema for a function'
+
+  static flags = {
+    ...cli.globalFlags,
+    ...appFlags,
+    'api-key': Flags.string({
+      name: 'API key',
+      description: 'The API key to fetch the schema with.',
+      required: false,
+    }),
+  }
+
+  static args = [{name: 'function', required: true, description: 'The name of the function to fetch the schema for.'}]
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(GenerateSchema)
+    const apiKey = flags['api-key']
+    const directory = flags.path ? path.resolve(flags.path) : process.cwd()
+    const app: AppInterface = await loadApp(directory)
+    const extension = app.extensions.function.find((extension) => extension.localIdentifier === args.function)
+
+    if (!extension) {
+      const functions = app.extensions.function.map((extension) => extension.localIdentifier).join(', ')
+
+      throw new error.Abort(
+        output.content`No function named ${args.function} found in this app.`,
+        output.content`Use one of the available functions: ${functions}`,
+      )
```

Comment:
@andrewhassan Looks fine.

Question, though — is there a way that we can prevent this error in the first place?

Reply:
Thanks!

>Question, though — is there a way that we can prevent this error in the first place?

I think the only way we can guarantee this value is valid is by having the user select the function from an interactive list. However, this makes it impossible for the command to be used in a non-interactive environment (e.g. CI), which is something we want to support. Since this means the value can be provided by the user, there's no way to validate it beforehand.